### ISSUE TITLE
Row#to_csv and Rows#to_csv should support header attribute

### ISSUE THREAD

**Issue by buren** [buren/honey_format#60]
tl;dr Add `#to_csv(headers: true)` option to `Rows` and `Row`. Do it by passing the `header` object from the `Matrix` class down to `Rows`.

__Longer__
Sometimes it would be really handy to be able to add column headers to `to_csv` when you only have access to `Rows` or `Row` class. Something like `row.to_csv(headers: true)`. For `Rows` its a bit more nuanced, if there are no rows the rows instance currently has no idea what the columns are. If there are we could do something ugly-ish like `rows.first&.members`, alternatively we could pass the header instance from the csv object down to rows.

See `Row#to_csv`  
https://github.com/buren/honey_format/blob/master/lib/honey_format/matrix/row.rb#L20
